### DISCUSSION TITLE
Auto-bail info?

### DISCUSSION THREAD

**Discussion by joaociocca** [epi052/feroxbuster#773] [General]
Ok, so I'm working on a script to automate a few things, kinda like making my own "autorecon" version, because reasons.

One thing I just noticed is there's no information, unless at least `-v` on auto-bail being triggered. For example, this command:

```
feroxbuster --stdin --insecure --random-agent --thorough --auto-bail --wordlist /usr/share/SecLists/Discovery/Web-Content/common-and-portuguese.txt --json --output "ferox-target.json" < urls
```

Spits out this output:

![image](https://user-images.githubusercontent.com/6473725/213732981-7d3654db-63aa-4208-a088-ed7b5c4e1c39.png)

When I ran it again with `-v`, I understood that there were no scans for the 3 URLs down there because auto-bail triggered. But without `-v`, there's no info at all. And since it sits at `0/x` it gives the impression that no scan was even attempted.

Also, that print is the end result, so it also seems like the two URLs where it got a really good progress just stopped out of nothing.

Trying and retrying with different levels of verbosity also showed me that most of the errors were for timeout, but there was another error, one `WRN      8.362 feroxbuster::scanner::requester Could not send FeroxResponse to output handler: channel closed`. 

Maybe, when auto-bail is on, we could have some of this info exposed on bail being triggered and why?

**Comment by joaociocca** [comment]
also, the "auto bail" on the splash is misaligned, missing a second space after its emoji =p

![image](https://user-images.githubusercontent.com/6473725/213772892-9d31f7c2-5990-4a22-80a6-c2e746cda152.png)

**Comment by epi052** [comment]
```
WRN      1.451 feroxbuster::scanner::requester too many Errors (25) triggered AutoBail Policy on running      http://127.0.0.1:8000/
```
right now i believe the only indicator is something like this in the logs; are you suggesting adding something in the primary output?

**Reply by joaociocca** [reply]
yep. Because unless we're actively looking for trouble, we won't turn verbose on, so would be nice to get a heads up, you know?

**Comment by epi052** [comment]
https://user-images.githubusercontent.com/43392618/214429395-4701cbb5-1aa0-4eb7-b6b5-cebde1c1c0ba.mp4

**Reply by joaociocca** [reply]
OHHHH I loved that new tuned rate message with the hand! =D

**Comment by epi052** [comment]
![image](https://user-images.githubusercontent.com/43392618/214430847-bbb664cd-b9dd-4439-8c50-b24d7d47d01e.png)

**Reply by joaociocca** [reply]
this is (chef's kiss) perfect!

**Comment by epi052** [comment]
glad you like them! these aren't final, but are pretty close to what'll land. I'll update again soon
